core/node/eth_sender/src/eth_tx_manager.rs

```diff
@@ -334,16 +334,22 @@ impl EthTxManager {
             // that `tx` is not mined and we should resend it.
             // We only resend the first un-mined transaction.
             if operator_nonce.latest <= tx.nonce {
+                let last_sent_at_block = storage
+                    .eth_sender_dal()
+                    .get_block_number_on_last_sent_attempt(tx.id)
```

Comment:
I'm a bit afraid of this approach, because we started to do more queries to L1 rpc provider.
The previous solution didn't have this problem

Reply:
I personally don't worry about it, it's indeed two queries per iteration more worst case, but only if we decreased tx_poll_period by a lot and it's currently set to 20 seconds on every env

Reply:
and this change will simplify code adding Gateway support as with it there isn't a single block numbers, but Gateway has different block number than L1